### ISSUE TITLE
Hello why this plugin that make such a large lag?

### ISSUE THREAD

**Issue by LSSwwwwwww** [Photon-GitHub/AACAdditionPro#473]
https://spark.lucko.me/V7Jwzr5grW
As you can see it.There are a lot of lag in this plugin.
Before I install this plugin.This protocolib is only 3-4.But install that.It has up to 13-16?
And the HoloExtension is also have lag.

This is the before CPU test.
https://spark.lucko.me/1hw9kbsNba

**Comment by Photon-GitHub** [comment]
According to the report AACAdditionPro takes only 4% of the server tick, most of it due to DamageIndicators. It is likely that Holomedals spawns a lot of entities that DamageIndicators has to process, which causes the lag.

**Comment by LSSwwwwwww** [comment]
Before delete the placehoder the spark is        https://spark.lucko.me/1RZWQm4OIa
After delete it the spark is https://spark.lucko.me/JyRRO8kROC
But how this AACAddotionPro can make the Protocolib 3% of cpu 0.0.Also it is a small performance problem.But I very hope you can Improve the performance of it.

**Comment by LSSwwwwwww** [comment]
https://spark.lucko.me/nIyvvcpooA

Well this will make the Protocolib have more performance.
So,If you can Please imporve it.
I test it on paperspigot-1.16.5
and there are 35 people in it.

**Comment by Janmm14** [comment]
what placeholder

**Comment by LSSwwwwwww** [comment]
Is it the placehoder's API's problem?

**Comment by Photon-GitHub** [comment]
This is mostly down to ProtocolLib from now on. I have profiled DamageIndicators myself, and the overwhelming majority of cpu time is now used to clone the packet. That is a ProtocolLib method and needed to have the health only clientsided.

I have requested a feature on spigot's issue tracker to do that without ProtocolLib, but I will only be able to use it, once implemented, on the newest version of minecraft. 

Until then I think I pretty much optimized DamageIndicators to the point where any improvements would be miniscule and increase maintenance time cost too much. Therefore I will close this ticket, feel free to ask any questions by tagging me.
